src/NextGen/DonationForm/Repositories/DonationFormRepository.php

```diff
@@ -0,0 +1,188 @@
+        $paymentDetails = $donationForm->getNodeByName('payment-details');
+
+        $paymentDetails->append(
+            Hidden::make('formId')
+                ->defaultValue($formId),
+
+            Hidden::make('currency')
+                ->defaultValue(give_get_currency($formId))
+        );
+
+        return $donationForm;
+    }
+
+    /**
+     * @unreleased
+     */
+    protected function convertFormBlocksToSection(stdClass $block): Section
+    {
+        return Section::make(substr($block->name, strpos($block->name, '/') + 1))
+            ->label($block->attributes->title)
+            ->description($block->attributes->description)
+            ->append(...array_map([$this, 'convertBlockToNode'], $block->innerBlocks));
+    }
+
+    /**
+     * @unreleased
+     * @throws EmptyNameException
+     */
+    protected function convertBlockToNode(stdClass $block): Node
+    {
+        if ($block->name === "custom-block-editor/donation-amount-levels") {
+            $node = Amount::make('amount')
+                ->levels(...array_map('absint', $block->attributes->levels))
+                ->allowCustomAmount()
+                ->defaultValue(50)
+                ->required();
+        } elseif ($block->name === "custom-block-editor/donor-name") {
+            $node = Name::make('name')->tap(function ($group) use ($block) {
+                $group->getNodeByName('firstName')
+                    ->label($block->attributes->firstNameLabel)
+                    ->placeholder($block->attributes->firstNamePlaceholder);
+
+                $group->getNodeByName('lastName')
+                    ->label($block->attributes->lastNameLabel)
+                    ->placeholder($block->attributes->lastNamePlaceholder)
+                    ->required($block->attributes->requireLastName);
+
+                if ($block->attributes->showHonorific) {
+                    $group->getNodeByName('honorific')
+                        ->label('Title')
+                        ->options(...$block->attributes->honorifics);
+                } else {
+                    $group->remove('honorific');
+                }
+            });
+        } elseif ($block->name === "custom-block-editor/paragraph") {
+            $node = Paragraph::make(substr(md5(mt_rand()), 0, 7))
+                ->content($block->attributes->content);
+        } elseif ($block->name === "custom-block-editor/email-field") {
+            $node = Email::make('email')->emailTag('email');
+        } elseif ($block->name === "custom-block-editor/payment-gateways") {
+            $node = PaymentGateways::make('gatewayId');
+        } elseif ($block->name === "custom-block-editor/donation-summary") {
+            $node = DonationSummary::make('donation-summary');
+        } elseif ($block->name === "custom-block-editor/company-field") {
+            $node = Text::make('company');
+        } else {
+            $node = Text::make($block->clientId);
+        }
```

Comment:
I think this could be cleaned up nicely if extracted into another function that uses the switch-return technique. We have a lot of `$node =` and `$block->name ===` that would be cleared up.

Reply:
Got it. @kjohnson Please make a note of this, then, as perhaps it fits your work.